### PR TITLE
v1.1.0

### PR DESCRIPTION
**Pull request by shubhw12**
- New: Introduced support for creating an individual checklist for custom post types.
- New: Added a list column to display the checklist status.
- New: Added a search bar to filter posts according to pending items.

### CODE REVIEW

File: classes/class-ppc-pagesetups.php [hunk 1]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+		/**
+		 * Dropdown for page.
+		 *
+		 * @since 1.1.0
+		 */
+		public function add_dropdown() {
+			$ppc_pst_type       = get_option( 'ppc_post_types_to_display' );
+			$ppc_checklist_data = PPC_Loader::get_instance()->get_list();
+			if ( isset( $ppc_pst_type ) && isset( $ppc_checklist_data ) ) {
+				foreach ( $ppc_pst_type as $key ) {
```
Comment (Nikschavan) [inline classes/class-ppc-pagesetups.php]: This `foreach` can be avoided by directly checking if current post_type exists in the option `ppc_post_types_to_display `

File: classes/class-ppc-pagesetups.php [hunk 2]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+					$ppc_page_data     = isset( $ppc_checklist_data[ $key ] ) ? $ppc_checklist_data[ $key ] : '';
+					$current_post_type = get_current_screen();
+					if ( $current_post_type->post_type === $key ) {
+						?>
+						<select class="select_multiple" name="slect_opt" id="slect_opt">
+							<option value=""><?php esc_html_e( 'Filter Unchecked...', 'pre-publish-checklist' ); ?></option>
+							<?php
+							if ( isset( $ppc_page_data ) && is_array( $ppc_page_data ) ) {
+								foreach ( $ppc_page_data as $ppc_key => $val ) {
```
Comment (Nikschavan) [inline classes/class-ppc-pagesetups.php]: Name the variables relevant to what they are storing. `$val `

File: classes/class-ppc-pagesetups.php [hunk 3]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+		public function ppc_add_column_name_func( $columns ) {
+			$columns = array(
+				'cb'            => '<input type="checkbox"/)',
+				'title'         => 'Title',
+				'author'        => 'Author',
+				'categories'    => 'Categories',
+				'tags'          => 'Tags',
+				'comments'      => '<span class="vers comment-grey-bubble" title="Comments"><span class="screen-reader-text">Comments</span></span>',
+				'date'          => 'Date',
+				'ppc_checklist' => 'Checklist Status',
```
Comment (Nikschavan) [inline classes/class-ppc-pagesetups.php]: The strings here should be translatable.

Comment (Nikschavan) [reply classes/class-ppc-pagesetups.php]: Also, We should just append out the key to array `$columns`. Many post types do not have `author`, `categories` etc. These will get added even if the post type does not support `author`, `categories` etc.

For eg check post type pages.

File: classes/class-ppc-pagesetups.php [hunk 4]
```diff
@@ -170,23 +333,25 @@ public function ppc_custom_box_html() {
...
+							<div class="ppc-checklist"><?php echo esc_attr( $ppc_value ); ?></div></div>
+						</label>
+						<?php
 				}
 				?>
-				<?php
+					<?php
 			} else {
-				echo 'Please create a list to display here from Settings->Pre-Publish-Checklist';
+				echo 'Please create a list to display. Click <a href="options-general.php?page=ppc&tab=ppc_general_settings" >here</a>';
```
Comment (Nikschavan) [inline classes/class-ppc-pagesetups.php]: The string should be translatable.

File: classes/class-ppc-pagesetups.php [hunk 5]
```diff
@@ -170,23 +333,25 @@ public function ppc_custom_box_html() {
...
-					<div class="ppc-checklist-item-wrapper">
-						<input type="checkbox" name="checkbox[]" id="<?php echo esc_attr( $ppc_key ); ?>" class="ppc_checkboxes" value= "<?php echo esc_attr( $ppc_key ); ?>" <?php empty( $value ) ? '' : checked( true, array_key_exists( $ppc_key, $value ) ); ?>
-						>
-					<div class="ppc-checklist"><?php echo esc_attr( $ppc_value ); ?></div></div>
-				</label>
-					<?php
+						<div class="ppc-checklist-item-wrapper">
+							<input type="checkbox" name="checkbox[]" id="<?php echo esc_attr( $ppc_key ); ?>" class="ppc_checkboxes" value= "<?php echo esc_attr( $ppc_key ); ?>" 
+							<?php
+							empty( $value ) ? '' : checked( true, array_key_exists( $ppc_key, $value ) );
```
Comment (Nikschavan) [inline classes/class-ppc-pagesetups.php]: 1. Rename the variable `$value` to be something relevant to the value it holds.
2. Check if ternary condition here can be avoided. The function `checked` does return empty string.

File: includes/ppc-checklist.php [hunk 6]
```diff
@@ -6,71 +6,109 @@
...
+							if ( ! empty( $ppc_checklist_item_data ) ) {
+								?>
+								<ul id="ppc-ul" class="ppc-ul"> 
+									<?php
+									foreach ( $ppc_checklist_item_data as $ppc_key => $ppc_value ) {
+										?>
+										<li class="ppc-li">
+											<span class="dashicons dashicons-menu-alt2 ppc-move-dashicon"></span> <input type="text" readonly="true" class="ppc-drag-feilds" $ppc_item_key ="<?php echo esc_attr( $ppc_key ); ?>" value="<?php echo esc_attr( $ppc_value ); ?>" name="ppc_checklist_item[]" >
+											<button type="button" id = "edit" name="Edit" class="ppcedit" value="<?php echo esc_attr( $ppc_key ); ?>"> <span class="dashicons dashicons-edit"></span>Edit</button>
+											<button type="button" id ="Delete" name="Delete" class="ppcdelete" value="<?php echo esc_attr( $ppc_key ); ?>"> <span class="dashicons dashicons-trash ppc-delete-dashicon"></span>Delete</button>
```
Comment (Nikschavan) [inline includes/ppc-checklist.php]: Make sure the strings are translatable.

File: classes/class-ppc-pagesetups.php (hunk 4 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Done

File: classes/class-ppc-update.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+	/**
+	 * PPC_Update initial setup
+	 *
+	 * @since 1.1.0
+	 */
```
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-update.php]: Corrected

File: classes/class-ppc-pagesetups.php [hunk 8]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+						?>
+						<select class="select_multiple" name="slect_opt" id="slect_opt">
+							<option value=""><?php esc_html_e( 'Filter Unchecked...', 'pre-publish-checklist' ); ?></option>
+							<?php
+							if ( isset( $ppc_page_data ) && is_array( $ppc_page_data ) ) {
+								foreach ( $ppc_page_data as $ppc_key => $val ) {
+									?>
+									<option value="<?php echo esc_attr( $ppc_key ); ?>" class="test"
+										<?php if ( isset( $_GET['slect_opt'] ) && wp_verify_nonce( sanitize_key( $_GET['slect_opt'] ) ) ) { ?>
+											<?php $type = wp_verify_nonce( sanitize_key( $_GET['slect_opt'] ) ); ?>
```
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Changes done.

File: classes/class-ppc-pagesetups.php [hunk 9]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+					$current_post_type = get_current_screen();
+					if ( $current_post_type->post_type === $key ) {
+						?>
+						<select class="select_multiple" name="slect_opt" id="slect_opt">
+							<option value=""><?php esc_html_e( 'Filter Unchecked...', 'pre-publish-checklist' ); ?></option>
+							<?php
+							if ( isset( $ppc_page_data ) && is_array( $ppc_page_data ) ) {
+								foreach ( $ppc_page_data as $ppc_key => $val ) {
+									?>
+									<option value="<?php echo esc_attr( $ppc_key ); ?>" class="test"
```
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: removed by pratiksha

File: classes/class-ppc-pagesetups.php (hunk 2 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Replaced with $filter_val

File: classes/class-ppc-pagesetups.php (hunk 3 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Translation done

File: classes/class-ppc-pagesetups.php (hunk 5 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Variable renamed.

File: includes/ppc-checklist.php (hunk 6 above)
Comment (Rahullakhwani) [reply includes/ppc-checklist.php]: Translation done

File: classes/class-ppc-update.php [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+		do_action( 'ppc_update_after' );
+	}
+
+	/**
+	 * Moves the option values from ppc_checklist_data to ppc_cpt_checklist_data.
+	 *
+	 * @since 1.1.0
+	 */
+	public function update_options_table() {
```
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-update.php]: Could you suggest a name, VP? 
In this function, we are changing the old 'ppc_checklist_data' to 'ppc_cpt_checklist_data' on plugin update.

File: classes/class-ppc-pagesetups.php (hunk 5 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: If I changed ternary operator `**empty( $value ) ? '' : checked( true, array_key_exists( $ppc_key, $value ) );**`  to **checked( true, array_key_exists( $ppc_key, $value ) );** it throws an error if the user does not check any item in post metabox.

File: classes/class-ppc-update.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (patilvikasj) [reply classes/class-ppc-update.php]: You can rename function name to someting link **migrate_data_to_new_option**

File: classes/class-ppc-pagesetups.php (hunk 5 above)
Comment (patilvikasj) [reply classes/class-ppc-pagesetups.php]: What error you are facing here?

Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: https://share.getcloudapp.com/rRuLkgX6

File: classes/class-ppc-pagesetups.php [hunk 11]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+		}
+		/**
+		 * For query.
+		 *
+		 * @param string $query for querying.
+		 *
+		 * @since 1.1.0
+		 */
+		public function posts_filter( $query ) {
+			if ( isset( $_GET['slect_opt'] ) && ! empty( $_GET['slect_opt'] ) ) {
```
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: && is_admin()
added

File: classes/class-ppc-update.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-update.php]: done

File: classes/class-ppc-pagesetups.php (hunk 5 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Retained:
empty( $value ) ? '' : checked( true, array_key_exists( $ppc_key, $value ) );

File: classes/class-ppc-pagesetups.php (hunk 1 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Removed foreach and checked with in_array function

File: classes/class-ppc-pagesetups.php [hunk 12]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+							<?php
+							if ( isset( $ppc_page_data ) && is_array( $ppc_page_data ) ) {
+								foreach ( $ppc_page_data as $ppc_key => $val ) {
+									?>
+									<option value="<?php echo esc_attr( $ppc_key ); ?>" class="test"
+										<?php if ( isset( $_GET['slect_opt'] ) && wp_verify_nonce( sanitize_key( $_GET['slect_opt'] ) ) ) { ?>
+											<?php $type = wp_verify_nonce( sanitize_key( $_GET['slect_opt'] ) ); ?>
+										<?php } ?>
+										<?php if ( isset( $_GET['slect_opt'] ) ) { ?>
+											<?php echo $_GET['slect_opt'] === $ppc_key ? 'selected' : ''; ?>
```
[the remark this answers is not in the document]
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Selected function used

File: classes/class-ppc-pagesetups.php (hunk 3 above)
Comment (pratikshasaman) [reply classes/class-ppc-pagesetups.php]: Extra columns removed. Replace with 
$columns['ppc_checklist'] = __( 'Checklist Status', 'pre-publish-checklist' );

**Pull request merged**